src/integration-test/java/com/commercetools/sync/integration/types/TypeSyncIT.java

```diff
@@ -0,0 +1,837 @@
+
+import static com.commercetools.sync.integration.commons.utils.ITUtils.deleteTypesFromTargetAndSource;
+import static com.commercetools.sync.integration.commons.utils.SphereClientUtils.CTP_SOURCE_CLIENT;
+import static com.commercetools.sync.integration.commons.utils.SphereClientUtils.CTP_TARGET_CLIENT;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.FIELD_DEFINITION_1;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.FIELD_DEFINITION_2;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.FIELD_DEFINITION_3;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.FIELD_DEFINITION_LABEL_1;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.FIELD_DEFINITION_NAME_1;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.TYPE_DESCRIPTION_1;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.TYPE_DESCRIPTION_2;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.TYPE_KEY_1;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.TYPE_KEY_2;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.TYPE_NAME_1;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.TYPE_NAME_2;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.getTypeByKey;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.populateSourceProject;
+import static com.commercetools.sync.integration.types.utils.TypeITUtils.populateTargetProject;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+public class TypeSyncIT {
+
+    /**
+     * Deletes types from source and target CTP projects.
+     * Populates source and target CTP projects with test data.
+     */
+    @Before
+    public void setup() {
+        deleteTypesFromTargetAndSource();
+        populateSourceProject();
+        populateTargetProject();
+    }
+
+    /**
+     * Deletes all the test data from the {@code CTP_SOURCE_CLIENT} and the {@code CTP_SOURCE_CLIENT} projects that
+     * were set up in this test class.
+     */
+    @AfterClass
+    public static void tearDown() {
+        deleteTypesFromTargetAndSource();
+    }
+
+
+    @Test
+    public void sync_WithUpdatedType_ShouldUpdateType() {
+        final Optional<Type> oldTypeBefore = getTypeByKey(CTP_TARGET_CLIENT, TYPE_KEY_1);
+        assertThat(oldTypeBefore).isNotEmpty();
```

Comment:
I don't think so because every test case working with the same data its just checks if any problem occurred in the @Before case. Even this lines is not required.

Reply:
Them just remove that line so it's not confusing.